SourceKitLSPDevUtils/Sources/ConfigSchemaGen/OptionDocument.swift

```diff
@@ -66,14 +66,34 @@ struct OptionDocumentBuilder {
           try appendProperty(property, indentLevel: indentLevel + 1)
         }
       case .enum(let schema):
-        for caseInfo in schema.cases {
-          // Add detailed description for each case if available
-          guard let description = caseInfo.description else {
-            continue
+        let hasAssociatedTypes = schema.cases.contains { $0.associatedProperties != nil && !$0.associatedProperties!.isEmpty }
+        
+        if hasAssociatedTypes {
+          let discriminatorFieldName = schema.discriminatorFieldName ?? "type"
+          doc += "\(indent)  - This is a tagged union discriminated by the `\(discriminatorFieldName)` field. Each case has the following structure:\n"
+          
+          for caseInfo in schema.cases {
+            doc += "\(indent)  - `\(discriminatorFieldName): \"\(caseInfo.name)\"`"
```

Comment:
Very nitpicky: I generally find that string literals that contain quotes read nicer when used in multi-line string literals because then you don’t need to escape the quotes.

```suggestion
            doc += """
              \(indent)  - `\(discriminatorFieldName): "\(caseInfo.name)"`
              """
```